pkg/util/util.go

```diff
@@ -164,6 +165,31 @@ func CurlEndpoint(endpoint string) ([]byte, error) {
 	return nil, fmt.Errorf("received a non-200 status when curling %s", endpoint)
 }
 
+func ValidateEndpoint(endpoint string) error {
+	var (
+		retries int = 3
+	)
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return fmt.Errorf("failed to parse the url: %v, err: %v", endpoint, err)
+	}
+
+	if len(u.Host) == 0 || len(u.Scheme) == 0 {
+		return fmt.Errorf("url %v is invalid", endpoint)
+	}
+
+	client := &http.Client{Timeout: 3 * time.Second}
```

Comment:
Will this fail for all private repos since we're not passing in the bearer token?

Reply:
this validation does not check for status code return. which means, as long as the URL is valid, it will pass. for example, for a private repo `https://github.com/yangcao77/multi-components-private`, it won't return any error for `http.get`, it's reachable, just the status code will be `404` without any token

Reply:
added private repo url to the unit test to avoid any confusion.

Reply:
Thanks for clarifying